File: backend/Models/Entities/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models.Entities
{
    [Table("bookings")]
    public class Booking
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        public long ClientId { get; set; }

        public long VendorId { get; set; }

        public string? VendorName { get; set; }

        public DateTime? EventDate { get; set; }

        public string? EventType { get; set; }

        public string Status { get; set; } = "PENDING";

        public double? TotalAmount { get; set; }

        public double? DepositAmount { get; set; }

        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: springboot-backend/src/main/java/com/wedspot/backend/Model/Entity/Booking.java
package com.wedspot.backend.Model.Entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "bookings")
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long clientId;
    private Long vendorId;
    private String vendorName;
    
    private LocalDateTime eventDate;
    private String eventType;
    private String status; // PENDING, CONFIRMED, COMPLETED, CANCELLED
    
    private Double totalAmount;
    private Double depositAmount;
    
    private String notes;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) status = "PENDING";
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
